pkg/provider/kvm.go

```diff
@@ -35,6 +35,25 @@ const KVMTemplate = `
       serial: true
       tpm-version: 2.0
       secure-boot: true
+      nics:
+      - device: virtio-net
+        id: nic0
+        network: user
+        ports:
+        - protocol: tcp
+          host:
+            address: ""
+            port: 22222
```

Comment:
I really need to implement the code to have the provider find an open port and an API for client to ask for VM's ssh port.